catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/AsyncFileAlterationObserverTest.java

```diff
@@ -0,0 +1,1007 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.apache.camel.spi.Synchronization;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Stubber;
+
+@RunWith(JUnit4.class)
+public class AsyncFileAlterationObserverTest {
+
+  private static String dummyData = "The duck may swim on the lake...";
+  private static String changedData = "the duck.";
+  private static int delayBuf = 125;
+
+  private Consumer<Runnable> doTestWrapper = Runnable::run;
+
+  private final AtomicInteger timesToFail = new AtomicInteger(0);
+
+  private int failures = 0;
+
+  private void doTest(Synchronization cb) {
+    synchronized (timesToFail) {
+      if (timesToFail.intValue() != 0) {
+        timesToFail.decrementAndGet();
+        failures++;
+        cb.onFailure(null);
+      } else {
+        cb.onComplete(null);
+      }
+    }
+  }
+
+  private File monitoredDirectory;
+  private AsyncFileAlterationObserver observer;
+
+  //  NESTED DIRECTORY VARS
+  private File childDir;
+  private File grandchildDir;
+  private File[] childFiles;
+  private File[] grandchildFiles;
+  private File[] files;
+  private File[] grandsiblingsFiles;
+
+  private AsyncFileAlterationListener fileListener;
+
+  private int totalSize;
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void setup() throws IOException {
+
+    fileListener = Mockito.mock(AsyncFileAlterationListener.class);
+    monitoredDirectory = temporaryFolder.newFolder("inbox");
+    observer = new AsyncFileAlterationObserver(monitoredDirectory);
+    observer.addListener(fileListener);
+
+    childDir = null;
+    grandchildDir = null;
+    childFiles = null;
+    grandchildFiles = null;
+    files = null;
+    grandsiblingsFiles = null;
+    totalSize = 0;
+
+    init();
+  }
+
+  private void init() {
```

Comment:
❓Why does this init method need to be called in the `@Before` method and also in every `@Test` method? Is there a simpler way to do this?

Reply:
I used it as a way to remove math. It's used a lot in the nested methods because I wanted to reset the creation count to 0 so anytime I created something I didn't have to assert `prior + new`